server/src/main/resources/db/changelog/20180110092945-cleanup-uebercerts-round4-premigration.xml

```diff
@@ -0,0 +1,100 @@
+
+    <property
+            dbms="postgresql,oracle,hsqldb"
+            name="round4_premigration_cleanup_ueber_ents"
+            value="delete from cp_entitlement where pool_id in (select id from cp_pool p where p.productname LIKE '%_ueber_product');" />
+
+    <property
+            dbms="postgresql,oracle,hsqldb"
+            name="round4_premigration_cleanup_ueber_pools"
+            value="delete from cp_pool where productname like '%_ueber_product%';" />
+
+    <property
+            dbms="postgresql,oracle,hsqldb"
+            name="round4_premigration_cleanup_ueber_content"
+            value="delete from cp_content where id in (select id from cp_content where name = 'ueber_content');" />
+
+    <property
+            dbms="postgresql,oracle,hsqldb"
+            name="round4_premigration_cleanup_ueber_products"
+            value="delete from cp_product where name like '%_ueber_product';" />
+
+    <property
+            dbms="postgresql,oracle,hsqldb"
+            name="round4_premigration_cleanup_ueber_consumers"
+            value="delete from cp_consumer where name = 'ueber_cert_consumer';" />
+
+    <!-- MySQL Queries -->
+    <property
+            dbms="mysql"
+            name="round4_premigration_cleanup_ueber_subs"
+            value="delete s from cp_subscription s inner join cp_product p on s.product_id = p.id where p.name LIKE '%_ueber_product';" />
+
+    <property
+            dbms="mysql"
+            name="round4_premigration_cleanup_ueber_ent_certs"
+            value="delete ec from cp_ent_certificate ec inner join cp_entitlement e on ec.entitlement_id = e.id inner join cp_pool p on e.pool_id = p.id where p.productname like '%_ueber_product';" />
+
+    <property
+            dbms="mysql"
+            name="round4_premigration_cleanup_ueber_ents"
+            value="delete e from cp_entitlement e inner join cp_pool p on e.pool_id = p.id where p.productname like '%_ueber_product';" />
+
+    <property
+            dbms="mysql"
+            name="round4_premigration_cleanup_ueber_pools"
+            value="delete p from cp_pool p where p.productname like '%_ueber_product';" />
+
+    <property
+            dbms="mysql"
+            name="round4_premigration_cleanup_ueber_content"
+            value="delete c from cp_content c
+               inner join cp_product_content pc on pc.content_id=c.id inner join cp_product p on p.id=pc.product_id where p.name LIKE '%_ueber_product';" />
+
+    <property
+            dbms="mysql"
+            name="round4_premigration_cleanup_ueber_products"
+            value="delete p from cp_product p where p.name LIKE '%_ueber_product';" />
+
+    <property
+            dbms="mysql"
+            name="round4_premigration_cleanup_ueber_consumers"
+            value="delete c from cp_consumer c where c.name = 'ueber_cert_consumer';" />
+
+    <changeSet id="20180110092945-1" author="vrjain">
```

Comment:
I'm seeing errors when I'm running this.  Here's my steps:

1. Deploy a clean Candlepin with `/bin/deploy -g`
2. Deploy Candlepin (so the update changelog runs) with `/bin/deploy`

I'm seeing errors like that `cp_subscription` doesn't exist and that `cp_pool` doesn't have a `productname` column.  I had to comment out a lot of statements just to test the delete element thing I proposed earlier.

Reply:
@vritant This is happening because during  liquibase DB creation the changeset that cleans up the old pre 2.0 tables / fields gets run (20170130144529-perorgproducts-phase-2.xml), but not your new one. On update, it sees that the new one hasn't yet been run yet and explodes.

You could probably use a precondition to get around this or add the changeset to the create changelog (though it seems redundant).

Reply:
@awood nice catch!
@mstead to me it seems cleaner to add preconditions, to do this during create seems like cheating. updated the code and added pre-condition, and verified this does not fail anymore.

also verified this PR still cleans up bad data even with the preconditions.

also added the pre-condition to the other branch as well.